qiskit/circuit/library/boolean_logic/inner_product.py

```diff
@@ -18,25 +18,54 @@
 
 
 class InnerProduct(QuantumCircuit):
-    """An n_qubit circuit that computes the inner product of two registers."""
+    r"""A 2n-qubit Boolean function that computes the inner product of
+    two n-qubit vectors over :math:`F_2`.
 
-    def __init__(self, num_qubits: int) -> None:
-        """Return a circuit to compute the inner product of 2 n-qubit registers.
+    This implementation is a phase oracle which computes the following transform.
+
+    .. math::
+
+        \mathcal{IP_2n} : F_2^{2n} \rightarrow {-1, 1}
+        \mathcal{IP_2n}(x_1, \cdots, x_n, y_1, \cdots, y_n) = (-1)^{x.y}
```

Comment:
Not 100% sure what should be in `mathcal` but I think the `2n` should be in curly braces to be subscript 🙂 
```suggestion
        \mathcal{IP}_{2n} : F_2^{2n} \rightarrow {-1, 1}
        \mathcal{IP}_{2n}(x_1, \cdots, x_n, y_1, \cdots, y_n) = (-1)^{x.y}
```